src/main/java/seedu/duke/MediTask.java

```diff
@@ -2,27 +2,73 @@
 
 import seedu.duke.commands.HospitalCommand;
 import seedu.duke.data.hospital.Hospital;
+import seedu.duke.ui.Ui;
+import seedu.duke.parser.Parser;
+import seedu.duke.data.hospital.Patient;
+import seedu.duke.data.state.State;
+import seedu.duke.data.state.StateType;
+import seedu.duke.commands.Command;
 
-import java.util.Scanner;
 
 public class MediTask {
     /**
      * Main entry-point for the java.duke.Duke application.
      */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-        System.out.println("What is your name?");
-
-        Scanner in = new Scanner(System.in);
-        System.out.println("Hello " + in.nextLine());
 
+        Ui ui = new Ui();
         Hospital hospital = new Hospital();
-        // Set hospital instance
+        ui.showWelcome();
+        // Start in MAIN_STATE
+        State currentState = new State(StateType.MAIN_STATE);
         HospitalCommand.setHospital(hospital);
+
+        // variable to hold the selected patient
+        Patient selectedPatient = null;
+
+        while (true) {
+            if (currentState.getState() == StateType.MAIN_STATE) {
+
+                ui.showMainScreen();
+                String commandInput = ui.readCommand();
+
+                // parse and execute the command, and also pass in  the currentState
+                Parser parser = new Parser(commandInput, currentState);
+                Command command = parser.parseCommand();
+
+                if (command != null) {
```

Comment:
The structure is a bit unclear here. Consider making the happy path more prominent to improve readability and flow.